### ISSUE TITLE
multiple word type in erDiagram attributes

### ISSUE THREAD

**Issue by marcosh** [mermaid-js/mermaid#3426]
As far as I understand it, when defining attributes in a `erDiagram`, both `type` and `name` need to be single words.

I am trying to define an `erDiagram` where I would like to use Haskell types instead of database types, to keep a more high-level vision.
The issue is that I have types like `Id Person`, which consists of two words and this breaks the parser.

The same issue would arise trying to use the [`integer ARRAY` syntax for Postgres](https://www.postgresql.org/docs/current/arrays.html#ARRAYS-DECLARATION).

Would it be possible to have a way to declare types with more than one word? Maybe using `"` or `{}` or whatever else is commonly used in Mermaid

**Comment by jgreywolf** [comment]
You could always use **_**.  Otherwise, enclosing the text in quotes could work as well.  Something to look at.

**Comment by Mikek16** [comment]
![Screenshot from 2023-11-26 22-54-54](https://github.com/mermaid-js/mermaid/assets/56452607/7520b07b-46dc-44e7-847b-d6cd49286f76)

It seems like declaring multi-word types/attributes with double quotes was added with PR #3516, should this issue be closed?

**Comment by f355** [comment]
@Mikek16 no, this is about entity FIELD types and names, not about the names of entities themselves, e.g. currently there's no way to make this snippet work:
```
erDiagram
  CUSTOMER {
    std::string name
  }
```

**Comment by Aditya-XR** [comment]
I checked this against current `develop`. Since #7786, attribute names and types can be wrapped in backticks, which covers the `std::string` example above:

```
erDiagram
  CUSTOMER {
    `std::string` name
  }
```

This parses fine now, and so does `varchar(255) email`. Unquoted `std::string name` still fails because `:` is part of the relationship syntax, but the backtick escape gives a way to write any multi-word or special-character type. I think this issue can be closed now, unless maintainers want the unescaped form to work too.
